[Infrastructure UI] Hosts table sorting and pagination

jennypavlova opened the pull request:
Closes https://github.com/elastic/kibana/issues/143510
## Summary

This PR adds pagination and column sorting to the hosts' table. I use default pagination settings (10 per page with the options to show 25 and 50). In order to avoid a change in the snapshot API for now I implemented it the same way as the inventory table view. If we see that the loading is slow and we need to optimize it we can change the snapshot API as well   
<img width="1448" alt="image" src="https://user-images.githubusercontent.com/14139027/196728634-b49a3abe-a5b7-4a03-8209-3e92ace8f0d9.png">

Testing:
I opened the inventory page in table view and compared the way pagination and sorting work there with the hosts' table using different sorting types ( ascending/descending order for each column ) 

Something I found out with the sorting while testing: (using `EuiInMemoryTable` ) When we have a text value (like the host.name) and we sort it (alphabetically for example) we will show first all the hosts starting with a capital letter (sorted alphabetically) and then the other hosts (sorted alphabetically) so the order will be for example **H**osta, Hostb, **a**host, bhost. It works the same way on the inventory page so I guess the sorting is implemented to be case-sensitive on purpose.

jennypavlova committed:
Sorting columns

jennypavlova committed:
Use EuiInMemoryTable pagination and sorting and change mapping

crespocarlos commented on x-pack/plugins/infra/public/pages/metrics/hosts/components/hosts_table.tsx:
```diff
@@ -18,9 +18,17 @@ interface Props {
 export const HostsTable: React.FunctionComponent<Props> = ({ nodes }) => {
   const items = useHostTable(nodes);
 
+  const sortSettings = {
```
I wonder whether we need to start the table sorting by cpuCores. Maybe moving forward it'd be nice to define what would be the best attribute here, or maybe leave the default api sorting, which currently is by @timestamp

jennypavlova replied:
I am not sure which element should be used by default and set the CPU cores but I can change it. The `@timestamp` field is not part of the table so maybe it will be a bit confusing to be used for sorting here (as visually there will be no indication inside the table header which column we use for sorting and the assumption is that it's not sorted), I think memory usage will be a good one. So as we are not sure if we need the default set here I will set `sorting={true}` to be the default and once we decide which column we want we can change it.

crespocarlos commented on x-pack/plugins/infra/public/pages/metrics/hosts/components/hosts_table_columns.tsx:
```diff
@@ -10,28 +10,29 @@ import React from 'react';
+  name: string;
 }
 
 export interface HostMetics {
```
I'm curious here. Why did we change the types here? Doesn't it work with the sortable table?

jennypavlova replied:
I changed the structure a bit - so for the sorting functionality we need a structure like `field_name: value (string/number)` and I was thinking that it's more readable if we have the name mapped separately (instead of using the label from the path)

crespocarlos commented on x-pack/plugins/infra/public/pages/metrics/hosts/hooks/use_host_table.ts:
```diff
@@ -6,17 +6,19 @@
       ...metrics.reduce((data, metric) => {
-        data[metric.name as keyof HostMetics] = metric;
+        data[metric.name as keyof HostMetics] =
+          metric.name === 'cpuCores' ? metric?.value : metric?.avg;
```
We need to be careful here. If we include a new field or use an attribute other than `avg` we will have to make sure it reflects here.

jennypavlova replied:
That's a good point! Most of the metrics we use currently are using `avg` and only the CPU cores are using the total number(value). Once we have the use case we can create the mapping separately for the fields we need ( I am not sure how the `services on host` structure will look like ) - so this mapping it definitely something we want to move separately once we know how the data will be structured.

jennypavlova replied:
I added a mapping to the values that should be used for each metric so we can extend it once we add more metrics: https://github.com/neptunian/kibana/pull/3/commits/0f49d15ed60d8c2bd65bb43b60da6c25b79c2960

crespocarlos reviewed (commented):
I'm curious about a couple of things around the usage of the `SnapshotNodeMetric` and the default sorting attribute. Left a few comments.

I'm assuming that we won't paginate the data on the api side, only on the FE, to prevent the page from growing too much, right?

Probably it's better to open this PR against main, once the table implementation has been merged.

jennypavlova commented:
> I'm assuming that we won't paginate the data on the api side, only on the FE, to prevent the page from growing too much, right?

Yes, that's correct. I discussed with Nathan that for now we won't implement the pagination in the snapshot api until we know how we want to do it and if we really need it there (we can also decide in the future to show only the top 100 hosts for example which won't require pagination at all) 
So I used the same idea we have in the inventory table view and used the eui in-memory table component to get the pagination out of the box on the FE.

> Probably it's better to open this PR against main, once the table implementation has been merged.

Yes, I opened it against the other PR so it's easier to see the diff here and to merge them together. But I agree, it will be better to wait for the other PR to be merged and I will update this one to be against main.

jennypavlova committed:
Set sorting to true and remove the default cpuCores field  sorting option

jennypavlova committed:
Update types

neptunian committed:
[Infrastructure UI] Replace Lens table with EUI table and own api (#142871)

* remove lens, add snapshot api

* add no data

* Add cpu type

* [WIP] Add eui basic table and columns

* [WIP] Add cpu cores and os

* [WIP] Mapping data to the eui basic table format

* Add Memory Total

* Refactor host mapping and add types

* Scale up memory usage percentage

* Make os optional in the model

* Text cells formatting

* Change os.type to os.name

* Move snapshot nodes mapping to a hook and test it

* Add translation

* Add fixed range and cleanup

* Fix diskLatency field name

* Remove not existing showQueryBar prop from SearchBar and set time range values

* Use last path

* Test last path change

* Type imports

* Change the way lastPath.os is set

* Type import

Co-authored-by: Jenny <dzheni.pavlova@elastic.co>

jennypavlova committed:
Merge branch 'main' into 143510-trying-hosts-table-sorting-and-pagination

jloleysens committed:
[Files] Filepicker (#143111)

* added unscoped file client to the files components context

* wip: created some basic files and stories for new filepicker component

* fix some types after require filesclient to be passed in

* added file picker state and some basic tests

* added file picker context

* added missing file client value

* updated file picker stories

* expanded files context;

* remove the size observable and also added a test for file removal and adding of duplicates

* added error content component

* refactor upload component to not take files client as prop

* updated shared file compoennts context value

* added test for adding multiple files at once

* allow passing in string or array of strings

* added some i18n texts

* move the creation of a file kinds registry to a common location

* set file kinds only once

* a bunch of stuff: added title component, using grid, removed responsive on upload controls

* refactor layour components to own component using grid, added new i18n texts for empty state prompt

* minor copy tweak

* added basic story with some files

* added file grid and refactored picker to only exist in modal for now

* get the css grid algo that we want: auto-fill not auto-fit!

* override styling for content area of file

* split stories of files

* delete commented out code

* give the modal a fixed width

* fix upload file state, where we do not want a fixed width modal

* moved styles down to card, and combined margin removal rules

* optimize for filtering files, first pass just filter on names

* include xxl

* moved debounceTime to rxjs land, added test for filtering behaviour

* added story with more images

* big ol wip

* empty prompt when uploading a file

* added pagination

* fixed tests and added some comments

* address lint

* moved copy to i18n and updated size and color of empty error prompt

* remove use of css`

* remove non existant prop

* also reload files

* fileUpload -> files

* update logic for watching if selected

* disambiguate i18n ids

* use abort signal and call sendRequest from file$, filtering done server side now, update tests

* fix a few off by one errors and hook up the new system to the ui

* added test for in flight requests behaviour

* update the files example app

* fix minor card layout styling to make all cards the same size regardless of text or image conten

* added new file picker component

* make file cards a bit wider and text a bit smaller so that it wraps...

* fix issue of throwing abort error and prematurely setting request to completed...

* remove unused import

* replace filter i18n

* a bunch of cool changes

* added export for the file picker component

* updated example app to use multiple file upload

* added some comments and made images load eagerly in file picker for now...

* complete ux for examples

* only files that are "READY" should be in the file picker

* set loading to false if error

* install data-test-subj everywhere!

* added some react component tests

* remove unused import

* fix storybook case

* fix up where the files example plugin is listed, moved it to the developer examples area

* fix potential flashing of loader by debouncing

* do not create new observable on every render

* i18n

* have only filepicker ctx used in filepicker components

* refactor loadImageEagerly -> lazy

* useObservable instead of useEffect

* factor modal footer to own component and remove css util

* use the middle dot luke

* copy update in files example app

* added filter story

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

jennypavlova committed:
Add mapping for the metric values used in hosts table

jennypavlova committed:
Merge branch 'main' into 143510-trying-hosts-table-sorting-and-pagination

jgowdyelastic committed:
[ML] Adding anomaly score explanations (#142999)

* [ML] Adding anomaly score explanations

* fixing types

* fixing jest test

* test title

* style improvements

* test data

* removing test data

* better test data

* removing test data

* [CI] Auto-commit changed files from 'node scripts/precommit_hook.js --ref HEAD~1..HEAD --fix'

* using eui colors

* reverting auto changes

* [CI] Auto-commit changed files from 'node scripts/precommit_hook.js --ref HEAD~1..HEAD --fix'

* tiny change

* reverting test data

* tiny change

* removing render functions

* adding tooltip text

* adding comments

* text update

* adding conditional impact score text

* text and scoring changes

* translations

* updating text

* more text changes

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

cnasikas committed:
[Cases] Remove alerts from cases (#143457)

* Remove alerts from cases

* Refactor property actions

* Add tests

* Fix i18n

* Fix tests

* PR feedback

* Do not show property action if empty

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

adcoelho committed:
Increase the max length limit of the case title to 160 characters  (#143664)

* Changed MAX_TITLE_LENGTH; fixed column width for case title.

* Updated tests with the new case title max length.

* Updated snapshots to fix failing tests.

* Increased the width of the select case modal in alerts.

shahzad31 committed:
[Exploratory view] Use ad hoc data views (#143358)

Commit:
[APM] Small perf improvement by omitting the source doc (#143743)

nickpeihl committed:
[Maps] Remove deprecated usage of indexPattern.title (#143444)

* Remove deprecated usage of indexPattern.title

* Fix mocks in unit tests

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

shahzad31 committed:
[Synthetics] Errors page added failed test count (#143633)

rockdaboot committed:
Support brotli compression on the server side (#142334)

* Use brotli compression

* [CI] Auto-commit changed files from 'node scripts/eslint --no-cache --fix'

* Add integration test for brotli support

* Use import instead of require()

* Suppress build error on importing brok

* [CI] Auto-commit changed files from 'node scripts/precommit_hook.js --ref HEAD~1..HEAD --fix'

* add brok as explicit package dep

* add `server.compression.brotli` config settings

* update documentation

* fix test utils

* fix more test configs

* add tests for endpoints too

* remove against endpoint for now

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: pgayvallet <pierre.gayvallet@elastic.co>

maximpn committed:
[Security Solution] Fix rule details enable/disable toggle isDisabled state (#143252)

**Resolves:** [#137149](https://github.com/elastic/kibana/issues/137149)

## Summary

Disables rule details enable/disable toggle for ML rules only under basic license.

Before:

https://user-images.githubusercontent.com/3775283/195514871-f0ccb25e-177d-4b4e-83bc-9c26da1718f0.mov

After:

https://user-images.githubusercontent.com/3775283/195513340-95944c6d-da6d-4ab3-9917-4e03f5791d3a.mov


### Checklist

- [x] Any UI touched in this PR does not create any new axe failures (run axe in browser: [FF](https://addons.mozilla.org/en-US/firefox/addon/axe-devtools/), [Chrome](https://chrome.google.com/webstore/detail/axe-web-accessibility-tes/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US))
- [x] This renders correctly on smaller devices using a responsive layout. (You can test this [in your browser](https://www.browserstack.com/guide/responsive-testing-on-local-server))
- [x] This was checked for [cross-browser compatibility](https://www.elastic.co/support/matrix#matrix_browsers)

kdelemme committed:
feat(slo): Support custom KQL indicator type (#143786)

alisonelizabeth committed:
[Guided onboarding] Fix continue button behavior (#143652)

jloleysens committed:
[Files] Upload design feedback + introduce compressed mode (#143744)

* use loading prop to drive button state

* changed upload complete state

* retry button should be EuiButton

* remove the success icon

* cancel button should use EuiButton

* add the compressed option

* make compressed mode!

* updated upload file story and component to always allow repeated uploads in compressed mode

* added error story

* added has files check

* added test for new #hasFiles

dokmic committed:
[Embeddable] Add seamless React integration (#143131)

* Add support of React nodes rendering to the embeddable panel
* Add support of React nodes rendering by the error handler
* Update visualizations embeddable to use React for error handling

dej611 committed:
[Lens] Random sampling feature (#143221)

* :sparkles: First pass with UI for random sampling

* :sparkles: Initial working version

* :fire: Remove unused stuff

* :wrench: Refactor layer settings panel

* :bug: Fix terms other query and some refactor

* :label: Fix types issues

* :bug: Fix sampling for other terms agg

* :bug: Fix issue with count operation

* :white_check_mark: Fix jest tests

* :bug: fix test stability

* :bug: fix test with newer params

* :lipstick: Add tech preview label

* :white_check_mark: Add new tests for sampling

* :white_check_mark: Add more tests

* :white_check_mark: Add new test for suggestions

* :white_check_mark: Add functional tests for layer actions and random sampling

* Update x-pack/plugins/lens/public/datasources/form_based/layer_settings.tsx

Co-authored-by: Michael Marcialis <michael@marcial.is>

* :ok_hand: Integrated design feedback

Co-authored-by: Joe Reuter <johannes.reuter@elastic.co>
Co-authored-by: Michael Marcialis <michael@marcial.is>

jennypavlova committed:
Merge branch 'main' into 143510-trying-hosts-table-sorting-and-pagination

KOTungseth committed:
Updates Edit data view link copy (#142883)

* Updates Edit data view link copy

* Updates Stack Management link text

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

Commit:
[APM] Make `track_total_hits` and `size` required in apm alerting client (#143851)

* [APM] Make `track_total_hits` and `size` required in apm alerting client

* undo changes

renovate[bot] committed:
Update babel (main) (#143490)

* Update babel

* dedupe

* update yarn.lock

* update snapshot

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Jonathan Budzenski <jon@elastic.co>

dokmic committed:
[Expressions] Fix the execution pipeline not to stop on a flaky subexpression (#143852)

* Fix the execution pipeline not to stop on a flaky subexpression
* Fix the execution pipeline not to stop on an invalid or incorrect value

demjened committed:
[ML Inference] Reorganize ML modules (#143803)

* Relocate delete and detach modules
* Relocate inference errors/history modules
* Relocate inference processors/history modules

lukasolson committed:
[KQL] Use term queries for keyword fields (#143599)

cqliu1 committed:
[Canvas] Use data views service (#139610)

* Use data views service in Canvas

* Use getIndexPattern instead of title property

* Fix ts errors

* Remove console log statement

drewdaemon committed:
[Lens] Metric trendlines design changes (#143781)

kc13greiner committed:
Removing docs about multi-tenancy (#143698)

* Removing docs about multi-tenancy

* Removing link to multi tenancy

nreese committed:
[Maps] Add ability to invert color ramp and size (#143307)

* convert color forms to TS

* switch for setting invert

* reverse colors

* convert size components to TS

* invert size switch

* invert mb size expression

* invert size legend

* invert ordinal legend

* invert colors in color picker

* update jest snapshot

* add unit tests for inverting color ramp creation

* review feedback

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

mistic committed:
skip flaky suite (#142222)

mistic committed:
skip flaky suite (#143907, #143908)

marshallmain committed:
[Security Solution][Alerts] Refactor rule execution logic integration tests (#142679)

* Separate rule execution logic tests and move bulk of the tests to preview for speed

* Remove bad dependency

* Update unit test snapshot

* Fix flaky test

* Fix another flaky test

* Fix more imports

* Remove superfluous return type

jbcrail committed:
[Profiling] Improve performance when decoding stacktraces (#143084)

* Derive address and file ID from base64 encoding

This skips the intermediate deserialization step to a buffer object.

* Move run-length encoding methods

* Decode run-length directly from base64 encoding

This skips the intermediate deserialization step to a buffer object.

* Minor refactor

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

kibanamachine committed:
[api-docs] Daily api_docs build (#143915)

mibragimov committed:
[Guided onboarding]  Support paragraph description for steps (#143831)

* [Guided onboarding] Support paragraph description for steps

* Add tests

Co-authored-by: Muhammad Ibragimov <muhammad.ibragimov@elastic.co>

stratoula committed:
[Unified search] Distinguishes the adhoc from the permanent dataviews on the pickers (#143525)

* In progress add icon to the adhoc picker

* [Unified search] Adds an icon to indicate that the selected dataview is not persisted

* Remove unused icon

* Add a temporary badge to the list

* Compare with the currentIndexpattern id

MiriamAparicio committed:
[APM] Extract apmEventClient from setupRequest (#143863)

* Extract apmEventClient from setupRequest

* continue refactor and fix tests

jennypavlova committed:
Merge branch 'main' into 143510-trying-hosts-table-sorting-and-pagination

walterra committed:
Call p-value aggs in parallel. (#143722)

In the analysis part we created chunks of 10 calls for the p-value aggregations, but individual calls of these chunks were then still called one after the other. This PR changes the behaviour to run calls for each chunk in parallel.

shahzad31 committed:
[Synthetics] Update data view permission test (#143921)

afharo committed:
[LaunchDarkly] Promote Cloud Experiments documentation in the Dev Docs (#143885)

LucaWintergerst committed:
[Profiling] allow negative numbers in formatter (#143420)

* Update as_number.ts

allow negative numbers in formatter. Currently all negative values get truncated to `~0.00`
closes https://github.com/elastic/prodfiler/issues/2707

* Update as_number.ts

* [CI] Auto-commit changed files from 'node scripts/precommit_hook.js --ref HEAD~1..HEAD --fix'

* Update as_number.ts

change negative condition to abs

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Tim Rühsen <tim.ruhsen@elastic.co>

yashtewari committed:
Edit Agent manifest to mount node's /etc directory. (#143492)

Corresponding change made in Agent repository with:
https://github.com/elastic/elastic-agent/pull/1382

Co-authored-by: Yash Tewari <yash.tewari@elastic.co>
Co-authored-by: Oren Zohar <85433724+oren-zohar@users.noreply.github.com>

nchaulet committed:
[Fleet] Fix error in agent policy duplication (#143884)

afharo committed:
[EBT] Report viewport size (#143854)

maximpn committed:
Fix invalid license info link (#143836)

**Relates to:** [#137148](https://github.com/elastic/kibana/issues/137148)

## Summary

Fixes an invalid link to the Platinum license info from the disabled ML rule card.

*Before:*

https://user-images.githubusercontent.com/3775283/197490117-b209250a-5fb9-49c8-bbd4-163aa649fd18.mov

*After:*

https://user-images.githubusercontent.com/3775283/197489478-ef7b428e-cc15-4f62-857b-1f53d5ae1325.mov


### Checklist

- [ ] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [x] Any UI touched in this PR is usable by keyboard only (learn more about [keyboard accessibility](https://webaim.org/techniques/keyboard/))
- [x] Any UI touched in this PR does not create any new axe failures (run axe in browser: [FF](https://addons.mozilla.org/en-US/firefox/addon/axe-devtools/), [Chrome](https://chrome.google.com/webstore/detail/axe-web-accessibility-tes/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US))
- [x] This renders correctly on smaller devices using a responsive layout. (You can test this [in your browser](https://www.browserstack.com/guide/responsive-testing-on-local-server))
- [x] This was checked for [cross-browser compatibility](https://www.elastic.co/support/matrix#matrix_browsers)

felixbarny committed:
Add missing options to APM central config (#143668)

kibanamachine committed:
skip failing test suite (#143109)

afharo committed:
Add Onboarding Setup Guide link in the help section when running on Cloud (#143765)

machadoum committed:
[Security Solution] Allow Users to launch Timeline from the Entity Analytics dashboard (#143841)

* Add alerts count column to Entity Analytics risk tables

jloleysens committed:
[Files] Fix image intersection observer behavior (#143843)

* react with loading spinner to new page or retry requests

* do not fetch on every change? in files example app

* remove lazy from the file picker image

* absolute position the image tag to ensure that it is visible, this ensures the intersection observer will fire

* remove the "lazy" prop from the image component

* remove unnecessary if

* added comment and increase page size of file picker in files example plugin

* make the empty "hidden" image occupy the full container space

dej611 committed:
:recycle: Unskip and fix tests (#143931)

Commit:
Cleanup RUM unsupported attrbiutes from base apm config. (#143937)

PhilippeOberti committed:
[TIP] Show EmptyPage only if no threat_intel integrations are installed (#143328)

[TIP] Show EmptyPage only if no threat_intel integrations are installed

- use fleet plugin's endpoint
- changed what is lazy loaded to make sure react-query isn't eagerly loaded

KOTungseth committed:
[Lens] Updates the Average and Count quick function definitions (#143872)

* Updates Edit data view link copy

* [Lens] Updates the Average and Count quick function definitions

* Reverts Stack Management change

* Removes spacing from Count definition

Commit:
refactor getHitCount, use Last_1 year (#143912)

* refactor getHitCount, use Last_1 year

* revert change to heartbeat test

kdelemme committed:
Add missing properties in saved object definition (#143796)

semd committed:
add apm tracking to matrixHistogram (#143950)

jennypavlova committed:
Merge branch 'main' into 143510-trying-hosts-table-sorting-and-pagination

guskovaue committed:
[RAM] 142183 create bulk delete on rules (#142466)

* create base for bulk delete route

* add bulk delete method

* add bulkDelete method for task manager

* first version of retry if conflicts for bulk delete

* move waitBeforeNextRetry to separate file

* add happy path for unit and integration tests

* rewrite retry_if_bulk_delete and add tests

* fix happy path integration test

* add some tests and fix bugs

* make bulk_delete endpoint external

* give up on types for endpoint arguments

* add unit tests for new bulk delete route

* add unit tests for bulk delete rules clients method

* add integrational tests

* api integration test running

* add integrational tests

* use bulk edit constant in log audit

* unskip skiped test

* fix conditional statement for taskIds

* api integration for bulkDelete is done

* small code style changes

* delete comments and rename types

* get rid of pmap without async

* delete not used part of return

* add audit logs for all deleted rules

* add unit tests for audit logs

* delete extra comments and rename constant

* delete extra tests

* fix audit logs

* restrict amount of passed ids to 1000

* fix audit logs again

* fix alerting security tests

* test case when user pass more that 1000 ids

* fix writing and case when you send no args

* fix line in the text

* delete extra test

* fix type for rules we passing to bulk delete

* add catch RuleTypeDisabledError

* wait before next retry func tests

* fix tests for retry function

* fix bulk delete rule client tests

* add to api return task ids failed to be deleted and wrap task manager call in try catch

* fix type for task manager

Co-authored-by: Xavier Mouligneau <xavier.mouligneau@elastic.co>
Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

mistic committed:
chore(NA): update versions after v7.17.8 bump (#143957)

jennypavlova committed:
Shorthand props

joeypoon committed:
[Security Solution] update generic policy response error (#143805)

jennypavlova committed:
Os from path - replace lodash and remove not needed null handling

walterra committed:
[ML] Explain Log Rate Spikes: Make use of abort signal for ES queries. (#143683)

So far we passed on the abort signal from the client to possibly cancel the analysis, but the signal was not passed on to the ES queries to cancel those. That means the analysis could be cancelled after each step but it did not cancel ES queries that were already running. This PR takes the already existing abort signal and passes it on to all ES queries.

This surfaced an issue with running too many queries in parallel: We didn't have a limit so far when fetching the histogram data. With using the abort signals, Kibana would report a warning if more than 10 queries were run at once. The PR updates fetching histogram data to also do it in chunks of 10 queries like we already do for the p-value aggregation. So the larger bulk of the file diff is the result of wrapping the histogram queries inside a for of to iterate over the chunks of queries.

maximpn committed:
[Security Solution] Fix Jira svg icon (#143916)

**Resolves:** [#140579](https://github.com/elastic/kibana/issues/140579)

## Summary

It replaces Jira svg icon with a simple one which works correctly in disabled mode.

## Details

The previous icon used an embedded png image which is applied through a `fill` property. It conflicts with the disabled mode which also uses `fill` property to display the icon in grey scale. An icon with embedded png images fixes it.

*Before:*

- Trial license 
<img width="986" alt="Screenshot 2022-10-25 at 09 17 21" src="https://user-images.githubusercontent.com/3775283/197698855-2a77579d-ea3d-45df-a10f-86105f0be300.png">

- Basic license
<img width="983" alt="Screenshot 2022-10-25 at 09 19 25" src="https://user-images.githubusercontent.com/3775283/197698877-0063c363-0bd6-4939-b23b-732828d4572a.png">

*After:*

- Trial license 
<img width="986" alt="Screenshot 2022-10-25 at 09 15 50" src="https://user-images.githubusercontent.com/3775283/197698913-922f921f-a379-4da7-97ce-10492f04ddbb.png">

- Basic license
<img width="987" alt="Screenshot 2022-10-25 at 09 20 24" src="https://user-images.githubusercontent.com/3775283/197698929-f7b6448e-fbf8-49e5-80f3-bc2af07d1847.png">

### Checklist

- [x] Any UI touched in this PR does not create any new axe failures (run axe in browser: [FF](https://addons.mozilla.org/en-US/firefox/addon/axe-devtools/), [Chrome](https://chrome.google.com/webstore/detail/axe-web-accessibility-tes/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US))
- [x] This renders correctly on smaller devices using a responsive layout. (You can test this [in your browser](https://www.browserstack.com/guide/responsive-testing-on-local-server))
- [x] This was checked for [cross-browser compatibility](https://www.elastic.co/support/matrix#matrix_browsers)

jennypavlova committed:
Fix types

jennypavlova committed:
Remove type export

abdonpijpelink committed:
Temporarily remove broken link (#143965)

Heenawter committed:
[Dashboard] [Controls] Add `excludes` toggle to options list (#142780)

* Add buttons with no functionality

* Added basic negate functionality

* Add `NOT` text when negated

* Clean up

* Add jest and functional tests

* Fix merge conflicts

* Rename `negate` to `exclude`

* Fix `unsaved changes` bug

* Move erase button back to beside search

* Clean up

* Add chaining functional tests

* Fix other unsaved changes bug

* Fix mobile view of popover

* Add option to disable exclude/include toggle

* Prevent unsaved changes bug for options list settings

* Add tooltip to run past timeout setting

* Address review comments

* Rename variable

* Set `exclude` to `false` when footer is hidden

machadoum committed:
Add warning message displayed for incompatible jobs in EA page (#143942)

* Add a warning message displayed for incompatible jobs on EA page

kertal committed:
[UnifiedFieldList] Remove kibana-vis-editors codeownership (#143953)

lukasolson committed:
Add link to acceptable formats in timeDefaults advanced setting (#143583)

* Add link to acceptable formats in advanced setting

* Translation goodness

* Update docs

jennypavlova committed:
Change metrics mapping

jennypavlova committed:
Merge branch 'main' into 143510-trying-hosts-table-sorting-and-pagination

animehart committed:
[Cloud Posture]Added local storage to remember pagination for 5 pages (#143835)

* added local storage to remember pagination for 5 pages

* addressing PR comments + Failed checks

* [CI] Auto-commit changed files from 'node scripts/precommit_hook.js --ref HEAD~1..HEAD --fix'

* addresing pr comments change pageSizes to pageSize

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

cauemarcondes committed:
[APM] Implement improvements to AWS Lambda metrics view (#143113)

* [APM] Refactoring metrics route

* apis

* refactoring memory function

* fixing merge conflicts

* refactoring

* [CI] Auto-commit changed files from 'node scripts/eslint --no-cache --fix'

* addin API test

* new settings

* cost settings

* removing cost settings

* refactoring api

* adding test

* Update oss_plugins.json

* refactoring apis

* new page

* fixing build

* metrics ui

* adding filter by serverlessFunctionName

* aws details

* refactoring

* fixing sorting

* tests

* test

* renaming api

* fixing test

* fixing ci

* refactoring

* reverting

* fixing breadcrumb

* addressing pr comments

* sanity check

* fixing merge conflicts

* fixing tests and ci

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

Heenawter committed:
[Dashboard] Fix help documentation link for dashboard  (#143894)

* Modify help link for dashboard

* Fix stub

jbudz committed:
Upgrade to Jest 27 (#139335)

* Upgrade to Jest 27

* fix test

byronhulcher committed:
Add clear button to domain validation input field (#143859)

byronhulcher committed:
Use target=_blank for links in validation steps (#143897)

byronhulcher committed:
Add copy functionality to domains table (#143785)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

YulNaumenko committed:
[Security Solution] Fixed flaky Jest config: x-pack/plugins/security_solution/public/timelines (#143824)

* [Security Solution] Fixed flaky Jest config: x-pack/plugins/security_solution/public/timelines

* Improved Ids

* -

* Fixed test files and lazy suricata

* Cleaned up act error for async stuff

* -

* Cleaning up jest errors for timeline components

* -

* Used mount enzyme

* mocked react-beautiful-dnd

* type check

* Fixed snapshot

* mocked one more expensive lib

* Reduced the number of data rows to 1 to simplify rendering

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

mitodrummer committed:
[Terminal Output] Truncated process output msg (#143304)

* xterm updated to v5. initial work done for max kb warning msg

* max bytes exceeded msg implemented, along with authz gated link to security policies page.

* marker indication on max bytes exceeded fixed

* tests written

* tests written

* fixed a bug where an active search would prevent normal playback/seeking

* [CI] Auto-commit changed files from 'node scripts/precommit_hook.js --ref HEAD~1..HEAD --fix'

* updates per review comments

* [CI] Auto-commit changed files from 'node scripts/eslint --no-cache --fix'

* fixed text

* fixed ts error

* fix to scroll hack

Co-authored-by: Karl Godard <karlgodard@elastic.co>
Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

jbudz committed:
skip flaky test. #143993

kibanamachine committed:
[api-docs] Daily api_docs build (#143997)

jloleysens committed:
[Files] Add files management privilege to `find` and `metrics` endpoints (#143274)

* added files management privileges definition and locked down metrics and find endpoint to management role

* [CI] Auto-commit changed files from 'node scripts/precommit_hook.js --ref HEAD~1..HEAD --fix'

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

cindychangy committed:
[Guided onboarding] Revised layout for panel footer (#143849)

* new layout for panel footer

* remove unused import

* remove footer bg color for darkMode

* replace custom icon styling with prop values

* move inline text style to prop

stratoula committed:
[Unified search] Change the button label when the query has changed (#143990)

jennypavlova committed:
Merge branch 'main' into 143510-trying-hosts-table-sorting-and-pagination
